kubeproxy/controllers/kubeproxy_controller.go

```diff
@@ -0,0 +1,216 @@
+	"sigs.k8s.io/kubebuilder-declarative-pattern/pkg/patterns/addon/pkg/status"
+	"sigs.k8s.io/kubebuilder-declarative-pattern/pkg/patterns/declarative"
+	"sigs.k8s.io/kubebuilder-declarative-pattern/pkg/patterns/declarative/pkg/manifest"
+
+	"github.com/go-logr/logr"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	api "addon-operators/kubeproxy/api/v1alpha1"
+)
+
+var _ reconcile.Reconciler = &KubeProxyReconciler{}
+
+// KubeProxyReconciler reconciles a KubeProxy object
+type KubeProxyReconciler struct {
+	client.Client
+	Log logr.Logger
+
+	declarative.Reconciler
+	watchLabels declarative.LabelMaker
+}
+
+func (r *KubeProxyReconciler) setupReconciler(mgr ctrl.Manager) error {
+	labels := map[string]string{
+		"k8s-app": "kubeproxy",
+	}
+
+	r.watchLabels = declarative.SourceLabel(mgr.GetScheme())
+
+	return r.Reconciler.Init(mgr, &api.KubeProxy{},
+		declarative.WithRawManifestOperation(injectFlags),
+		declarative.WithRawManifestOperation(replaceNamespacePattern("{{.Namespace}}")),
+		declarative.WithObjectTransform(declarative.AddLabels(labels)),
+		declarative.WithObjectTransform(OverrideApiserver),
+		declarative.WithOwner(declarative.SourceAsOwner),
+		declarative.WithLabels(r.watchLabels),
+		declarative.WithStatus(status.NewBasic(mgr.GetClient())),
+		declarative.WithObjectTransform(addon.TransformApplicationFromStatus),
+		declarative.WithManagedApplication(r.watchLabels),
+		declarative.WithObjectTransform(addon.ApplyPatches),
+	)
+}
+
+// +kubebuilder:rbac:groups=addons.x-k8s.io,resources=kubeproxies,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=addons.x-k8s.io,resources=kubeproxies/status,verbs=get;update;patch
+
+func (r *KubeProxyReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	addon.Init()
+
+	if err := r.setupReconciler(mgr); err != nil {
+		return err
+	}
+
+	c, err := controller.New("kubeproxy-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to KubeProxy
+	err = c.Watch(&source.Kind{Type: &api.KubeProxy{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to deployed objects
+	_, err = declarative.WatchAll(mgr.GetConfig(), c, r, r.watchLabels)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// replaceNamespacePattern fills in the namespace placeholder patterns with the actual namespace from the crd
+func replaceNamespacePattern(nspatterns ...string) declarative.ManifestOperation {
+	return func(ctx context.Context, o declarative.DeclarativeObject, manifest string) (string, error) {
+		for _, pattern := range nspatterns {
+			if strings.Index(manifest, pattern) != -1 {
+				manifest = strings.Replace(manifest, pattern, o.GetNamespace(), -1)
+			}
+		}
+		return manifest, nil
+	}
+}
+
+func injectFlags(ctx context.Context, object declarative.DeclarativeObject, s string) (string, error) {
```

Comment:
Almost all command line flags of kube-proxy are now deprecated (see [here](https://github.com/kubernetes/kubernetes/blob/1fa20301a0173eb7945ea8b80ac60997adceaffa/cmd/kube-proxy/app/server.go#L226)). Hence, I would advocate for the use of the kube-proxy component config here.

cc @mtaufen @stealthybox